research/gaia/gremlin/gs_gremlin/src/bin/start_rpc_server.rs

```diff
@@ -0,0 +1,108 @@
+//! you may not use this file except in compliance with the License.
+//! You may obtain a copy of the License at
+//!
+//! http://www.apache.org/licenses/LICENSE-2.0
+//!
+//! Unless required by applicable law or agreed to in writing, software
+//! distributed under the License is distributed on an "AS IS" BASIS,
+//! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//! See the License for the specific language governing permissions and
+//! limitations under the License.
+
+use gremlin_core::compiler::GremlinJobCompiler;
+use gremlin_core::register_gremlin_types;
+use gs_gremlin::store::global_storage::create_gs_store;
+use gs_gremlin::store::global_storage::MultiPartition;
+use gs_gremlin::store::mg_mock::build_modern_mock_graph;
+use log::info;
+use pegasus::Configuration;
+use pegasus_server::config::combine_config;
+use pegasus_server::rpc::start_debug_rpc_server;
+use pegasus_server::service::Service;
+use pegasus_server::{CommonConfig, HostsConfig};
+use std::sync::Arc;
+use structopt::StructOpt;
+
+#[derive(Debug, Clone, StructOpt)]
+#[structopt(about = "An RPC server for accepting queries from Gremlin server.")]
+pub struct ServerConfig {
+    #[structopt(
+        long = "port",
+        short = "p",
+        default_value = "1234",
+        help = "the port to accept RPC connections"
+    )]
+    pub rpc_port: u16,
+    #[structopt(
+        long = "index",
+        short = "i",
+        default_value = "0",
+        help = "the current server id among all servers"
+    )]
+    pub server_id: u64,
+    #[structopt(
+        long = "hosts",
+        short = "h",
+        default_value = "",
+        help = "the path of hosts file for pegasus communication"
+    )]
+    pub hosts: String,
+    #[structopt(
+        long = "config",
+        short = "c",
+        default_value = "",
+        help = "the path of config file for pegasus"
+    )]
+    pub config: String,
+    #[structopt(
+        long = "report",
+        help = "the option to report the job latency and memory usage"
+    )]
+    pub report: bool,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    pegasus_common::logs::init_log();
+    let server_config: ServerConfig = ServerConfig::from_args();
+    let host_config = if server_config.hosts.is_empty() {
+        None
+    } else {
+        Some(HostsConfig::read_from(server_config.hosts)?)
+    };
+    let common_config = if server_config.config.is_empty() {
+        None
+    } else {
+        Some(CommonConfig::read_from(server_config.config)?)
+    };
+    let num_servers = if let Some(h) = &host_config {
+        h.peers.len()
+    } else {
+        1
+    };
+    let config = combine_config(server_config.server_id, host_config, common_config);
+    let addr = format!("{}:{}", "0.0.0.0", server_config.rpc_port);
+
+    let graph = build_modern_mock_graph();
+    let partitioner = graph.clone();
```

Comment:
Not sure what this means, `partitioner = graph.clone()`? Why a partitioner is cloning a graph?

Reply:
This is just an example (and will be replaced when integrate with graphscope storage). I have implemented GraphPartitionManager trait (from graphscope) for MockGraph, following the way  graphscope storage does it.

Reply:
If this is the case, a recommendation is to mark this pull request as [WIP], and use this temporary MockGraph in your own test instead of submitting to the code bases. I've tried to review the codes, but MockGraph-related codes introduce too many distractions (and to be honest, I find it hard to understand some codes).

Reply:
Ok, I have removed the start_rpc_server and the mock_graph related files, which are for test.